### PR TITLE
Support `/api/v1/admin/domain_blocks`

### PR DESCRIPTION
**Pull request by JacksonSamuel42**

### CODE REVIEW

**Comment by neet** [comment]
@JacksonSamuel42 Is this ready for review? https://github.com/neet/masto.js/pull/739#discussion_r1038331906 and https://github.com/neet/masto.js/pull/739#pullrequestreview-1202781663 are not seemed to be fixed

> It should be a singular DomainBlock

> Masto.js internally converts property names from snake_case to camelCase to comply with JS conventions, so please update all snake_cases to annotate types properly

**Comment by JacksonSamuel42** [comment]
@neet sorry I forget to check

**Comment by JacksonSamuel42** [comment]
all changes implemented

File: src/entities/admin/domain-block.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  /** The date of the application that created this account. */
+  severity: SeverityType;
+  /** The reject media of the domain. */
+  rejectMedia: boolean;
+  /** The reject report of the domain. */
+  rejectReposts: boolean;
+  /** The private comment of the domain. */
+  privateComment?: null;
+  /** The public comment of the domain. */
+  publicComment?: null;
```
Comment (neet) [inline src/entities/admin/domain-block.ts:19]: Do you mean `string | null` ?

File: src/repositories/admin/domain-blocks-repository.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  readonly severity?: SeverityType;
+  /** Whether media attachments should be rejected*/
+  readonly reject_media?: boolean;
+  /** Whether reports from this domain should be rejected*/
+  readonly reject_reports?: boolean;
+  /**  A private note about this domain block, visible only to admins*/
+  readonly private_comment?: boolean | null;
+  /** A public note about this domain block, optionally shown on the about page*/
+  readonly public_comment?: string | null;
+  /** Whether to partially censor the domain when shown in public*/
```
Comment (neet) [inline src/repositories/admin/domain-blocks-repository.ts:20]: You still have snake_cases

File: src/repositories/admin/domain-blocks-repository.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  readonly obfuscate?: boolean;
+}
+
+export type AllBlockedDomain = {
```
Comment (neet) [inline src/repositories/admin/domain-blocks-repository.ts:24]: I could not understand what `AllBlockedDomain` is for by its name. It should be `FetchAllBlockedDomainParams` or something like that.

File: .eslintrc.json [hunk 4]
```diff
@@ -20,7 +20,8 @@
     "no-console": "error",
     "simple-import-sort/imports": "error",
     "unicorn/prevent-abbreviations": "off",
-    "unicorn/no-array-reduce": "off"
+    "unicorn/no-array-reduce": "off",
+    "prettier/prettier": ["error", { "endOfLine": "auto" }, {"usePrettier": true}]
```
Comment (neet) [inline .eslintrc.json:24]: What is this change for? I've been using `lf` as EOL and changing it to `auto` does not seem to make any effect.
https://prettier.io/docs/en/options.html#end-of-line

**Review by neet — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 4 above)
Comment (JacksonSamuel42) [reply .eslintrc.json]: image

and all files have this error but i remove this line and run a script into packege.json to solve all files

**Pull request merged**